include/fluent-bit/flb_engine_macros.h

```diff
@@ -52,4 +53,25 @@
 #define FLB_ENGINE_TASK         2
 #define FLB_ENGINE_IN_CORO      3
 
+/* Engine priority queue configuration */
+#define FLB_ENGINE_LOOP_MAX_ITER        10 /* Max events processed per round */
+
+/* Engine event priorities: min value prioritized */
+#define FLB_ENGINE_PRIORITY_DEFAULT     MK_EVENT_PRIORITY_DEFAULT
+#define FLB_ENGINE_PRIORITY_COUNT       8
+#define FLB_ENGINE_PRIORITY_TOP         0
+#define FLB_ENGINE_PRIORITY_BOTTOM      (FLB_ENGINE_PRIORITY_COUNT - 1)
+#define FLB_ENGINE_PRIORITY_NETWORK     1
+
+#define FLB_ENGINE_PRIORITY_CB_SCHED    FLB_ENGINE_PRIORITY_TOP
+#define FLB_ENGINE_PRIORITY_CB_TIMER    FLB_ENGINE_PRIORITY_TOP
+#define FLB_ENGINE_PRIORITY_SHUTDOWN    FLB_ENGINE_PRIORITY_TOP
+#define FLB_ENGINE_PRIORITY_FLUSH       FLB_ENGINE_PRIORITY_TOP
+
+#define FLB_ENGINE_PRIORITY_DNS         FLB_ENGINE_PRIORITY_NETWORK
+#define FLB_ENGINE_PRIORITY_CONNECT     FLB_ENGINE_PRIORITY_NETWORK
+#define FLB_ENGINE_PRIORITY_SEND_RECV   FLB_ENGINE_PRIORITY_NETWORK
+
+#define FLB_ENGINE_PRIORITY_THREAD      FLB_ENGINE_PRIORITY_DEFAULT
```

Comment:
@edsiper would you please review these priorities carefully? Here's a summary:
1. Most prioritized: priority 0, Flush (task/coroutine dispatch - new system for engine) and all callbacks
2. Medium priority: priority 1, Networking events
3. Least prioritized: Default priority, Coroutine/task initialization